internal/processor/aggregator.go

```diff
@@ -0,0 +1,85 @@
+package processor
+
+import (
+	"github.com/go-gota/gota/dataframe"
+	"github.com/go-gota/gota/series"
+)
+
+func Aggregate_data(df dataframe.DataFrame) dataframe.DataFrame {
+	df = df.FilterAggregation(
+		dataframe.And,
+		dataframe.F{Colname: "owner_kind", Comparator: series.Neq, Comparando: ""},
+		dataframe.F{Colname: "owner_name", Comparator: series.Neq, Comparando: ""},
+		dataframe.F{Colname: "workload", Comparator: series.Neq, Comparando: ""},
+		dataframe.F{Colname: "workload_type", Comparator: series.Neq, Comparando: ""},
+	)
+
+	columns := df.Names()
+	index_of_owner_name := findInStringSlice("owner_name", columns)
+	index_of_owner_kind := findInStringSlice("owner_kind", columns)
+	index_of_workload := findInStringSlice("workload", columns)
+	index_of_workload_type := findInStringSlice("workload_type", columns)
+
+	s := df.Rapply(func(s series.Series) series.Series {
+		owner_name := s.Elem(index_of_owner_name).String()
+		owner_kind := s.Elem(index_of_owner_kind).String()
+		workload := s.Elem(index_of_workload).String()
+		workload_type := s.Elem(index_of_workload_type).String()
+		if owner_kind == "ReplicaSet" && workload == "<none>" {
+			return series.Strings([]string{"replicaset", owner_name})
+		} else if owner_kind == "ReplicationController" && workload == "<none>" {
+			return series.Strings([]string{"replicationcontroller", owner_name})
```

Comment:
Shall we use the WorkloadType enum values here?

Reply:
Done